workflow/rules/utils.smk

```diff
@@ -36,32 +36,76 @@ rule tabix_known_variants:
         "0.56.0/bio/tabix"
 
 
-rule get_covered_regions:
+rule map_primers:
     input:
-        ref="resources/genome.fasta",
-        bam="results/recal/{sample}.sorted.bam",
-        bai="results/recal/{sample}.sorted.bam.bai"
+        reads=[config["primers"]["trimming"]["primers_fa1"], config["primers"]["trimming"]["primers_fa2"]],
+        idx=rules.bwa_index.output
     output:
-        temp("results/regions/temp/{sample}.quantized.bed.gz")
+        "results/mapped/primers.bam"
+    log:
+        "logs/bwa_mem/primers.log"
     params:
-        prefix=lambda wc, output: output[0].split(".quantized.bed.gz")[0]
-    shadow: "minimal"
+        index=lambda w, input: os.path.splitext(input.idx[0])[0],
+        sort="samtools",
+        sort_order="coordinate",
+        extra="-T 10 -k 8 -c 5000"
+
+    threads: 8
+    wrapper:
+        "0.56.0/bio/bwa/mem"
+
+
+rule get_primer_insert:
+    input:
+        "results/mapped/primers.bam",
+        "results/mapped/primers.bam.bai"
+    output:
+        "results/primers/primers.txt"
+    log:
+        "logs/primers/primers.log"
+    conda:
+        "../envs/pysam.yaml"
+    script:
+        "../scripts/extract_primers_insert.py"
+
+
+rule get_primer_interval:
+    input:
+        "results/mapped/primers.bam"
+    output:
+        "results/primers/target_regions.bed"
     log:
-        "logs/bam-regions/{sample}.log"
+        "logs/primers/target_regions.log"
     conda:
-        "../envs/mosdepth.yaml"
+        "../envs/bedtools.yaml"
     shell:
-        "mosdepth {params.prefix} {input.bam} -q 1: 2> {log}"
+        "samtools sort -n {input} | bamToBed -i - -bedpe | "
+        "awk '{{print $1 \"\t\" $2 \"\t\" $6}}' | "
+        "sort -k1,1 -k2,2n | mergeBed -i - > {output}"
+
+
+rule build_genome_bed:
+    input:
+        "resources/genome.fasta.fai"
+    output:
+        "results/regions/genome_regions.bed"
+    params:
+        chroms=config["ref"]["n_chromosomes"]
+    log:
+        "logs/regions/genome_regions.log"
+    script:
+        "../scripts/fasta_generate_genome.py"
```

Comment:
Please choose a more speaking name.

Reply:
build-genome-bed? Isn't fai almost a bed?

Reply:
maybe this can be a one liner with xsv

Reply:
I would prefer using `cut` as this exists by default and we don't need add another dependancy.

Reply:
Additionally, I moved this rule into `build_excluded_regions` as it just reads existing information of the input.